### ISSUE TITLE
gevent error when locust server started and browse the website

### ISSUE THREAD

**Issue by SQZ777** [locustio/locust#1142]
### Describe the bug
When I started the locust server then I browse the website, the log is:

```
python -m locust
[2019-11-13 00:24:30,516] SQZ777/INFO/locust.main: Starting web monitor at *:8089
[2019-11-13 00:24:30,517] SQZ777/INFO/locust.main: Starting Locust 0.12.2
[2019-11-13 00:24:33,820] SQZ777/ERROR/stderr: <gevent._socket3.socket object, fd=956, family=23, type=1, proto=0>: Invalid HTTP method: '\x16\x03\x01\x02\x00\x01\x00\x01ü\x03\x03Yç\x95ç.d\x9aÿN$¾Ò\x03>þ\\îÅî9³\t\x8e\x90\x8fe\x18J¥D\x8cú Ê*Øî\x08[\x01í\x9c\xad©tá »\x04vðìè\x15E\x96\x8cêXaõCrö\x81\x00"zz\x13\x01\x13\x02\x13\x03À+À/À,À0Ì©Ì¨À\x13À\x14\x00\x9c\x00\x9d\x00/\x005\x00\n'
[2019-11-13 00:24:33,822] SQZ777/ERROR/stderr: <gevent._socket3.socket object, fd=948, family=23, type=1, proto=0>: Invalid HTTP method: '\x16\x03\x01\x02\x00\x01\x00\x01ü\x03\x03\x9c¦d_à\x8b[xý\x9eúaþÒ&\x11m\x00hCºY8ô\x05\x13$úF\t\x98\x94 \x13Ý7J*\x9dÂ)\x9dq_ÐNêÂ¼\x80\x8fáñ\x01·2\x11É\x1f6hKUpÎ\x00"\x1a\x1a\x13\x01\x13\x02\x13\x03À+À/À,À0Ì©Ì¨À\x13À\x14\x00\x9c\x00\x9d\x00/\x005\x00\n'
[2019-11-13 00:24:33,851] SQZ777/ERROR/stderr: <gevent._socket3.socket object, fd=960, family=23, type=1, proto=0>: Invalid http version: '\x16\x03\x01\x02\x00\x01\x00\x01ü\x03\x03)p,.ò\x8d¸\x96ñ\x91þª¸,ûÏØ@<ÆRßÏBº!Ø\x80ñûöÂ cb\x1dLÄ\'\x0fÖG\x84gvm±G\x975$ÏêÎï7\x10\x07sØ»*i_\x17\x00"\x8a\x8a\x13\x01\x13\x02\x13\x03À+À/À,À0Ì©Ì¨À\x13À\x14\x00\x9c\x00\x9d\x00/\x005\x00\n'
```

### Environment settings

- OS: Windows 10
- Python version: 3.7.5
- Locust version: 0.12.2

**Comment by heyman** [comment]
Please post a minimal reproducible code example.

**Comment by SQZ777** [comment]
```from locust import Locust, TaskSet, task

class MyTaskSet(TaskSet):
    @task
    def my_task(self):
        print("executing my_task")

class MyLocust(Locust):
    task_set = MyTaskSet
    min_wait = 5000
    max_wait = 15000```

**Comment by heyman** [comment]
That code works perfectly fine for me (once I add imports), so it sounds like there's some issue with your environment.

**Comment by lolwhitaker** [comment]
@lundstrj I just debugged this same error message with someone and it turned out they had a config error.  They has set the worker's master-port setting to use the WEB_UI port (8089) instead of using the expected/default bind port (5557).  Not sure why this was the error in that case, but it went away when they corrected that config issue.

**Comment by lundstrj** [comment]
@lolwhitaker thanks for reaching out. I haven't change my worker's master-port but I do have the default 8089 for the web_ui. I'll see if I can locate the setting for the worker master port and see if fiddling with it will help.

What I don't get is why this problem temporarily goes away after I reboot, only to resurface again a little bit later in the day. Searching doesn't yield much but I wonder if it could be related to certificates perhaps?

**Comment by lundstrj** [comment]
`❯ locust -f locker.py -u 1 -r 1 --master-port=5557
[2021-11-01 20:10:55,007] Riscy.local/INFO/locust.main: Starting web interface at http://0.0.0.0:8089 (accepting connections from all network interfaces)
[2021-11-01 20:10:55,012] Riscy.local/INFO/locust.main: Starting Locust 2.4.1
<gevent._socket3.socket at 0x107bbca00 object, fd=8, family=30, type=1, proto=0>: Invalid HTTP method: "\x00\x00z\x00\x00\x00\x00\x04\x00\x00\x01\x00\x00\x00\x00'\x10\x7fÿÿÿ\n"
<gevent._socket3.socket at 0x107bbc8e0 object, fd=8, family=30, type=1, proto=0>: Invalid HTTP method: "\x00\x00z\x00\x00\x00\x00\x04\x00\x00\x01\x00\x00\x00\x00'\x10\x7fÿÿÿ\n"
<gevent._socket3.socket at 0x107bbc8e0 object, fd=8, family=30, type=1, proto=0>: Invalid HTTP method: "\x00\x00z\x00\x00\x00\x00\x04\x00\x00\x01\x00\x00\x00\x00'\x10\x7fÿÿÿ\n"`

Assuming I used that flag correctly, it didn't fix my problem =/

**Comment by lundstrj** [comment]
@heyman Sorry to bother you but maybe we could reopen this and try to work together to find a fix? I'd be happy to try things and provide data if it could help in anyway?

**Comment by cyberw** [comment]
See if you can reproduce it with just gevent + requests (it is unlikely that it is caused by locust itself). Maybe they have seen similar issues..

As a workaround, you can also try using FastHttpUser instead and see if it works better.

**Comment by cyberw** [comment]
I’ll re-close this soon if there are no more details.

**Comment by lundstrj** [comment]
@cyberw Hello, I'm sorry, I missed that you reopened this (no notification for some reason).

I was using FastHttpUser already. I could have tried to switch to the regular HttpUser and see if that has an effect.

I have done a fair bit of searching for gevent issues that look similar to this one and the closest I could find (naturally I've lost the link) hinted that it could be a certificate related issue (Python does not use the system certificates but instead has it's own certificate handling).

I tried any and all suggested solutions related to running straight up requests (without gevent) without having to set `verify=False` to make the ssl errors/warning to go away (unsuccessfully I might add).

Then one day these `gevent._socket3.socket` warnings just stopped. Perhaps my attempts at fixing my certificate issue with `certifi` kicked in after a reboot or something. Perhaps a `pip install --upgrade locust` pulled down a newer versions of gevent (or some other library) which resolved this.

Unfortunately I don't know what I did (if anything) that caused this to stop occurring (but stop it did) and that does bother me a bit.

**Comment by cyberw** [comment]
Thanks for the update!

**Comment by StevenBorg** [comment]
I just ran into this issue today. Basically, nothing gets sent once this error occurs on master. Here's an actual output from the logs. I get about 3 of these during startup.
`<gevent._socket3.socket at 0x7fbd0d1a49a0 object, fd=22, family=10, type=1, proto=0>: Invalid http version: '\x16\x03\x00\x00i\x01\x00\x00e\x03\x03U\x1c§ärandom1random2random3random4\x00\x00\x0c\x00/\x00\n'`

Since I'm running in Azure Container Instances, I took the 'reboot' fix advice above and simply redeployed my infrastructure. That fixed it on the first go. 

Using Locust 2.1.0, but will likely upgrade now that I'm seeing this issue.  (I hadn't changed any code or my underlying container, so not sure why this popped up.)
